src/main/java/cloud/eppo/api/AssignmentDetails.java

```diff
@@ -0,0 +1,31 @@
+package cloud.eppo.api;
+
+/**
+ * Contains both the assigned variation value and comprehensive evaluation details explaining why
+ * that variation was assigned.
+ *
+ * @param <T> The type of the variation value (Boolean, Integer, Double, String, etc.)
+ */
+public class AssignmentDetails<T> {
```

Comment:
This is the top-level class that will be returned by `get<type>Details()` methods.

We use generics so that the customer's code doesn't have to unwrap an EppoValue if they retrieve the variation (which is top level, typed T).